lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/AzureBatchEvaluatorShimManager.java

```diff
@@ -0,0 +1,339 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.reef.runime.azbatch.driver;
+
+import com.microsoft.azure.batch.BatchClient;
+import com.microsoft.azure.batch.auth.BatchSharedKeyCredentials;
+import com.microsoft.azure.batch.protocol.models.ResourceFile;
+import com.microsoft.azure.batch.protocol.models.TaskAddParameter;
+import org.apache.reef.annotations.audience.DriverSide;
+import org.apache.reef.annotations.audience.Private;
+import org.apache.reef.proto.EvaluatorShimProtocol;
+import org.apache.reef.runime.azbatch.evaluator.EvaluatorShimConfiguration;
+import org.apache.reef.runime.azbatch.parameters.AzureBatchAccountKey;
+import org.apache.reef.runime.azbatch.parameters.AzureBatchAccountName;
+import org.apache.reef.runime.azbatch.parameters.AzureBatchAccountUri;
+import org.apache.reef.runime.azbatch.util.AzureStorageUtil;
+import org.apache.reef.runime.azbatch.util.CommandBuilder;
+import org.apache.reef.runime.azbatch.util.RemoteIdentifierParser;
+import org.apache.reef.runtime.common.driver.api.ResourceLaunchEvent;
+import org.apache.reef.runtime.common.driver.api.ResourceReleaseEvent;
+import org.apache.reef.runtime.common.driver.api.ResourceRequestEvent;
+import org.apache.reef.runtime.common.driver.evaluator.pojos.State;
+import org.apache.reef.runtime.common.driver.resourcemanager.NodeDescriptorEventImpl;
+import org.apache.reef.runtime.common.driver.resourcemanager.ResourceEventImpl;
+import org.apache.reef.runtime.common.driver.resourcemanager.RuntimeStatusEventImpl;
+import org.apache.reef.runtime.common.files.*;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.annotations.Parameter;
+import org.apache.reef.tang.exceptions.BindException;
+import org.apache.reef.tang.formats.ConfigurationSerializer;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.runtime.common.utils.RemoteManager;
+import org.apache.reef.wake.remote.RemoteMessage;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.apache.reef.runime.azbatch.driver.RuntimeIdentifier.RUNTIME_NAME;
+
+/**
+ * Evaluator Shim Manager class.
+ */
+@Private
+@DriverSide
+public final class AzureBatchEvaluatorShimManager
+    implements EventHandler<RemoteMessage<EvaluatorShimProtocol.EvaluatorShimStatusProto>> {
+
+  private static final Logger LOG = Logger.getLogger(AzureBatchEvaluatorShimManager.class.getName());
+
+  private static final String AZ_BATCH_JOB_ID_ENV = "AZ_BATCH_JOB_ID";
+
+  private static final String EVALUATOR_SHIM_CONF_FILE = "shim.config";
```

Comment:
We probably should align the naming to shim.conf

Reply:
yep, thank you!